pydantic/fields.py

```diff
@@ -373,12 +373,12 @@ class MyModel(pydantic.BaseModel):
             init_var = False
             if annotation is dataclasses.InitVar:
                 init_var = True
-                annotation = Any
+                annotation = typing.cast(Any, Any)
```

Comment:
This is a really weird behavior from pyright, where the following would happen:

```python
from typing import Any

var = Any

reveal_type(var)  # Revealed type is "Any"

def func(arg: type): pass

func(var)  # Type error
```

The reason is that `var` is inferred as being the actual `typing.Any` object, and not the `Any` "typing concept" (but pyright displays it the same way).

This also relates to the Typing form PEP, that would be of great help here.

Reply:
Hmph, I was definitely confused by this before reading the explanation. Do you think this is something that might change on the pyright end of things any time soon?

Reply:
I'll open a discussion on the pyright repo to see if we can find a better workaround

Reply:
Unfortunately we are using some undefined behavior here (`Any` is meant to be used as an annotation, not as a value), so I don't expect pyright to special case things here :/ But this clearly shows the need for a TypeForm feature in the typing system!

Reply:
I can probably add a comment in the code in the meanwhile

Reply:
@Viicos,

That'd be great. Yeah, I realize now that a lot of the changes here just showcase the utility of a `TypeForm` feature :).